PetGame/Models/LeaderboardRequestModel.cs

```diff
@@ -0,0 +1,11 @@
+﻿namespace PetGame.Models
+{
+    /// <summary>
+    /// Represents the request body for the Leaderboard API
+    /// </summary>
+    public class LeaderboardRequestModel
+    {
+        public int Offset { get; set; }
+        public int NumRequests { get; set; }
```

Comment:
Rename `NumRequests` to `NumItems`. You are only making a single request, not many. Name is misleading.